edx_solutions_api_integration/mobile_api/views.py

```diff
@@ -15,3 +23,26 @@ class MobileUsersOrganizationsList(MobileListAPIView, UsersOrganizationsList):
     """
     def __init__(self):
         self.permission_classes += (IsStaffOrOwner, )
+
+
+class MobileUsersDiscussionMetrics(MobileListAPIView, UsersSocialMetrics):
+    """
+    View to return user discussion metrics and engagement score.
+    """
+    def get(self, request):
+        course_id = self.request.query_params.get('course_id', None)
+        username = self.request.query_params.get('username', None)
+        course_key = get_course_key(course_id)
+
+        if not course_exists(request, self.request.user, course_id):
+            raise Http404
+
+        user = self.request.user
+        if not user.is_staff and not CourseEnrollment.is_enrolled(user, course_key):
+            raise PermissionDenied
+        if username is not None:
+            if user.username != username and not user.is_staff:
+                raise PermissionDenied
+            user = get_object_or_404(User, username=username)
```

Comment:
Making sure I understand this as it's similar to what I am doing for the Proficiency API. If a staff member specifies a username and a course the user is not enrolled in, the checks in here will pass then the helper methods in `UsersSocialMetrics` will end up with a mismatch and it will probably return some score like 0 (I think), when a 404 with an informative message would probably be more helpful. Is it something we want to worry about or are we trusting requests to at least get that right?

Reply:
@BogdanL You're right, I missed that. It should return a 404 in that case.